lighthouse-core/audits/jankless-font.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview
+ * Audit that checks whether fonts that use `font-display: optional` were preloaded.
+ */
+
+const Audit = require('./audit.js');
+const i18n = require('./../lib/i18n/i18n.js');
+const FontDisplay = require('./../audits/font-display.js');
+const PASSING_FONT_DISPLAY_REGEX = /^(optional)$/;
+const NetworkRecords = require('../computed/network-records.js');
+
+const UIStrings = {
+  /** Title of a Lighthouse audit that provides detail on whether fonts that used `font-display: optional` were preloaded. This descriptive title is shown to users when all fonts that used `font-display: optional` were preloaded. */
+  title: 'Fonts with `font-display: optional` are preloaded',
+  /** Title of a Lighthouse audit that provides detail on whether fonts that used `font-display: optional` were preloaded. This descriptive title is shown to users when one or more fonts used `font-display: optional` and were not preloaded. */
+  failureTitle: 'Fonts with `font-display: optional` are not preloaded',
+  /** Description of a Lighthouse audit that tells the user why they should preload fonts if they are using `font-display: optional`. This is displayed after a user expands the section to see more. No character length limits. 'Learn More' becomes link text to additional documentation. */
+  description: 'Preloading optional fonts can reduce layout shifts and improve CLS. [Learn More](https://web.dev/optimize-cls/#web-fonts-causing-foutfoit)',
```

Comment:
```suggestion
  description: 'Preloading optional fonts can reduce layout shifts and improve CLS. [Learn More](https://web.dev/preload-optional-fonts/)',
```

better link right?

Reply:
Good point! I had `web.dev/preload-optional-fonts/` in the first commit, then changed it because I thought it was important to capture that this audit was relevant to CLS (and `web.dev/optimize-cls` has a link to `web.dev/preload-optional-fonts/`), but according to #11227, preload doesn't affect CLS; however, if the scope of this audit changes according to #11227, maybe `web.dev/optimize-cls` is worth keeping since it is more general?

Regardless, I'll change it back to `web.dev/preload-optional-fonts/` for now :)

Reply:
ya. as it turns out neither is perfect, but i'd go with houssein's article for now.